extensions/wikia/ImageReview/modules/PromoteImage/PromoteImageReviewTask.php

```diff
@@ -1,429 +1,429 @@
+		$this->mVisible = false; // do not show form for this task
+		$this->mParams = $params;
+		$this->mTTL = 86400; // 24 hours
+		$this->records = 1000; // TODO: needed?
+		$this->initializeSkippedWikiList();
+
+		parent::__construct();
+	}
+
+	protected function initializeSkippedWikiList() {
+		$cityVisualization = new CityVisualization();
+		$corporateSites = $cityVisualization->getVisualizationWikisData();
+
+		$dbNames = array();
+		foreach ($corporateSites as $site) {
+			$dbNames[] = $site['db'];
+		}
+
+		$this->dbNamesToBeSkipped = $dbNames;
+	}
+
+	function execute($params = null) {
+		$this->mTaskID = $params->task_id;
+		$oUser = User::newFromId($params->task_user_id);
+
+		if( $oUser instanceof User ) {
+			$oUser->load();
+			$this->mUser = $oUser->getId();
+		} else {
+		//if task was added by import script it has WikiaBot user's id passed
+			$this->log(__CLASS__ . ' / ' . __METHOD__ . ": Invalid user #1 - task_user_id: " . $params->task_user_id);
+			$oUser = User::newFromName('WikiaBot');
+
+			if( $oUser instanceof User ) {
+				$oUser->load();
+				$this->mUser = $oUser->getId();
+			} else {
+				$this->log(__CLASS__ . ' / ' . __METHOD__ . ": Invalid user #2 - task_user_id: " . $params->task_user_id);
+			}
+
+			return true;
+		}
+
+		/** @var WikiGetDataForVisualizationHelper $this->helper  */
+		$this->helper = new WikiGetDataForVisualizationHelper();
+		/** @var WikiGetDataForVisualizationHelper $this->model  */
+		$this->model = new CityVisualization();
+		$this->corporatePagesIds = $this->model->getVisualizationWikisIds();
+
+		$data = unserialize($params->task_arguments);
+		if( isset($data['upload_list']) ) {
+			foreach( $data['upload_list'] as $targetWikiId => $wikis ) {
+				$uploadResult = $this->uploadImages($targetWikiId, $wikis);
+			}
+		}
+
+		if( isset($data['deletion_list']) ) {
+			foreach( $data['deletion_list'] as $corpWikiLang => $wikis ) {
+				$deleteResult = $this->removeImages($corpWikiLang, $wikis);
+			}
+		}
+
+		if( !empty($uploadResult) || !empty($deleteResult) ) {
+			return true;
+		}
+
+		return false;
+	}
+
+	function finalizeImageUploadStatus($imageId, $sourceWikiId, $status){
+		$db = wfGetDB(DB_MASTER, array(), F::app()->wg->ExternalSharedDB);
+
+		$db->update(
+			'city_visualization_images',
+			array(
+				'reviewer_id = null',
+				'image_review_status' => $status,
+			),
+			array(
+				"city_id " => $sourceWikiId,
+				"page_id" => $imageId,
+				'image_review_status' => ImageReviewStates::APPROVED_AND_TRANSFERRING,
+			),
+			__METHOD__
+		);
+	}
+
+	/**
+	 * @desc This method uploads images from a wiki to corporate wiki (i.e. wikia.com or de.wikia.com) but also it can upload images from corporate wiki to target wikis
+	 *
+	 * @param $targetWikiId
+	 * @param $wikis
+	 * @return bool
+	 */
+	function uploadImages($targetWikiId, $wikis) {
+		$isError = false;
+
+		$targetWikiLang = WikiFactory::getVarValueByName('wgLanguageCode', $targetWikiId);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "wgLanguageCode" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)